src/haz3lcore/dynamics/EvaluatorStep.re

```diff
@@ -1163,7 +1163,12 @@ module Decompose = {
         decompose(d1) >>| Return.wrap(c => Cast(c, ty, ty'))
       | BoundVar(_) => Return.mark(act)
       | Sequence(d1, d2) =>
-        decompose(d1) >>| Return.wrap(c => Sequence(c, d2))
+        let* r1 = decompose(d1);
+        switch (r1) {
+        | BoxedValue
+        | Eval(_) => Return.mark(act)
+        | _ => r1 |> Return.wrap(c => Sequence(c, d2)) |> Monad.return
```

Comment:
generally good not to use a wildcard to make future refactoring (e.g. adding new cases) easier -- list out the cases

Reply:
I use _ pattern in other places - let me remove those patterns altogether.

Reply:
This switch statement actually only has one input, so there's only 4 possibilities. I've written them all out now.